src/modules/esl-share/core/esl-share-config.ts

```diff
@@ -31,32 +35,137 @@ export interface ESLShareGroupConfig {
   list: string;
 }
 
-/** The definition of `ESLShare` component configuration */
-export interface ESLShareConfig {
-  /** List of sharing buttons configuration */
-  buttons: ESLShareButtonConfig[];
-  /** List of share button groups configurations */
-  groups: ESLShareGroupConfig[];
-}
-
+/** Gets item of the section by name */
 function getConfigSectionItem<T extends ESLShareButtonConfig | ESLShareGroupConfig>(section: T[], name: string): T | undefined {
   return section.find((item) => item.name === name);
 }
 
-/**
- * Selects the buttons for the given list and returns their configuration.
- * @returns config of buttons
- */
-export function selectButtonsForList(config: ESLShareConfig, list: string): ESLShareButtonConfig[] {
-  return list.split(' ').reduce((res, item) => {
-    const [btnName, groupName] = item.split('group:');
-    if (groupName) {
-      const groupConfig = getConfigSectionItem(config.groups, groupName);
-      if (groupConfig) return res.concat(selectButtonsForList(config, groupConfig.list));
-    } else {
-      const btnConfig = getConfigSectionItem(config.buttons, btnName);
-      if (btnConfig) res.push(btnConfig);
+/** Inserts item into section or updates if exists */
+function setConfigSectionItem<T extends ESLShareButtonConfig | ESLShareGroupConfig>(section: T[], item: T): void {
+  const index = section.findIndex((i) => i.name === item.name);
+  if (index >= 0) section[index] = item;
+  else section.push(item);
+}
+
+/** Class for managing share buttons configurations */
+export class ESLShareConfig extends SyntheticEventTarget {
+  @memoize()
+  public static get instance(): ESLShareConfig {
+    return new ESLShareConfig();
```

Comment:
Cosmetic: possibly we need to declare constructor private/protected then

Reply:
done